myosuite/utils/heightfields.py

```diff
@@ -0,0 +1,410 @@
+                    while terrain_type.name == 'HILLY' and generated_terrains[TerrainTypes.HILLY.value] >= 2:
+                        terrain_type = self.rng.choice(TerrainTypes)
+                    generated_terrains[terrain_type.value] += 1
+                    self._fill_patch(i, j, terrain_type)
+            # put special terrain only once in 20% of episodes
+            if self.rng.uniform() < 0.2:
+                i, j = np.random.randint(0, self.patches_per_side, size=2)
+                self._fill_patch(i, j, SpecialTerrains.RELIEF)
+
+    def _fill_patch(self, i, j, terrain_type=TerrainTypes.FLAT):
+        """
+        Fill patch at position <i> ,<j> with terrain <type>
+        """
+        self.hfield.data[i * self.patch_size: i*self.patch_size + self.patch_size,
+                    j * self.patch_size: j * self.patch_size + self.patch_size] = self._compute_patch_data(terrain_type)
+        
+    def sample(self, rng=None):
+        """
+        Sample an entire heightfield for the episode.
+        Update geom in viewer if rendering.
+        """
+        if not rng is None:
+            self.rng = rng
+        self._populate_patches()
+        if hasattr(self.sim, 'renderer') and not self.sim.renderer._window is None:
+            self.sim.renderer._window.update_hfield(0)
+
+    # Patch types  ---------------
+
+    def _compute_rough_terrain(self):
+        """
+        Compute data for a random noise rough terrain.
+        """
+        rough = self.rng.uniform(low=-1.0, high=1.0, size=(self.patch_size, self.patch_size))
+        normalized_data = (rough - np.min(rough)) / (np.max(rough) - np.min(rough))
+        scalar, offset = .08, .02
+        scalar = self.rng.uniform(low=self.rough_range[0], high=self.rough_range[1])
+        return normalized_data * scalar - offset
+
+    def _compute_relief_terrain(self):
+        """
+        Compute data for a special logo terrain.
+        """
+        curr_dir = os.path.dirname(__file__)
+        relief = np.load(os.path.join(curr_dir, '../assets/myo_relief.npy'))
+        normalized_data = (relief - np.min(relief)) / (np.max(relief) - np.min(relief))
+        return np.flipud(normalized_data) * self.rng.uniform(self.relief_range[0], self.relief_range[1])
+
+    def _compute_hilly_terrain(self):
+        """
+        Compute data for a terrain with smooth hills.
+        """
+        frequency = 10
+        scalar = self.rng.uniform(low=self.hills_range[0], high=self.hills_range[1])
+        data = np.sin(np.linspace(0, frequency * np.pi, self.patch_size * self.patch_size) + np.pi / 2) - 1
+        normalized_data = (data - data.min()) / (data.max() - data.min())
+        normalized_data = np.flip(normalized_data.reshape(self.patch_size, self.patch_size) * scalar, [0, 1]).reshape(self.patch_size, self.patch_size)
+        if self.rng.uniform() < 0.5:
+            normalized_data = np.rot90(normalized_data)
+        return normalized_data
+    
+
+class TrackField(HeightField):
```

Comment:
Why is this in utils?

Reply:
where else should we move it? I want a common heightfield file for all environments that use them

myochallenge from last year and this year subclass the same heightfield class

Reply:
Yes, that's the right idea. Maybe 
- myosuite/envs 
- myosuite/envs/myo 
might be better places for it